cdap-ui/app/cdap/components/Experiments/store/ExperimentDetailActionCreator.js

```diff
@@ -207,134 +155,134 @@ function getSplitsInExperiment(experimentId) {
-    if (matchingHyperParameter) {
-      return matchingHyperParameter.label;
+  experimentDetailsStore.dispatch({
+    type: EXPERIMENTDETAILACTIONS.SET_MODELS_LOADING,
+    payload: {
+      modelsLoading
     }
-    return hyperparam;
-  }
-  return hyperparam;
-};
+  });
+}
 
-const setAlgorithmsList = () => {
-  let algoList = AlgorithmsListStore.getState();
-  if (algoList.length) {
-    return;
+function addModelsWithError(modelId) {
+  let modelsWithError = [...experimentDetailsStore.getState().modelsWithError];
+  if (modelsWithError.indexOf(modelId) === -1) {
+    modelsWithError.push(modelId);
   }
-  myExperimentsApi
-    .getAlgorithms({
-      namespace: getCurrentNamespace()
-    })
-    .subscribe(algorithmsList => {
-      algorithmsList = algorithmsList.map(algo => ({...algo, name: algo.algorithm}));
-      AlgorithmsListStore.dispatch({
-        type: AlgorithmsStoreActions.SET_ALGORITHMS_LIST,
-        payload: {algorithmsList}
-      });
-    });
-};
-
-function updateQueryParameters({ limit, offset, sortMethod, sortColumn }) {
-  experimentsStore.dispatch({
-    type: ACTIONS.SET_QUERY_PARAMS,
+  experimentDetailsStore.dispatch({
+    type: EXPERIMENTDETAILACTIONS.SET_MODELS_WITH_ERROR,
     payload: {
-      limit,
-      offset,
-      sortMethod,
-      sortColumn
+      modelsWithError
     }
   });
 }
 
-function updateQueryString() {
-  let {
-    offset,
-    limit,
-    sortMethod,
-    sortColumn
-  } = experimentsStore.getState().experiments;
-  let newQuery = `offset=${offset}&limit=${limit}&sort=${sortColumn} ${sortMethod}`;
-  let obj = {
-    title: document.title,
-    url: `${location.pathname}?${newQuery}`
-  };
-  history.pushState(obj, obj.title, obj.url);
-}
-
-function handlePageChange({selected}) {
-  let {limit} = experimentsStore.getState().experiments;
-  experimentsStore.dispatch({
-    type: ACTIONS.SET_PAGINATION,
+function removeModelsWithError(modelId) {
+  let modelsWithError = [...experimentDetailsStore.getState().modelsWithError];
+  let modelIndex = modelsWithError.indexOf(modelId);
+  if (modelIndex !== -1) {
+    modelsWithError.splice(modelIndex, 1);
+  }
+  experimentDetailsStore.dispatch({
+    type: EXPERIMENTDETAILACTIONS.SET_MODELS_WITH_ERROR,
     payload: {
-      offset: selected * limit
+      modelsWithError
     }
   });
-  updateQueryString();
-  getExperimentsList();
 }
 
-function handleExperimentsSort(field) {
-  let { sortColumn, sortMethod } = experimentsStore.getState().experiments;
-  let newSortField = (field !== sortColumn) ? field : sortColumn;
-  let newSortMethod = MMDS_SORT_METHODS.ASC === sortMethod ? MMDS_SORT_METHODS.DESC : MMDS_SORT_METHODS.ASC;
-  experimentsStore.dispatch({
-    type: ACTIONS.SET_EXPERIMENTS_SORT,
+function setActiveModel(activeModelId) {
+  let state = experimentDetailsStore.getState();
+  let {splitDetails} = state.models.find(model => model.id === activeModelId);
```

Comment:
You might want to guard the find with an empty object. Otherwise we might get a runtime error for not able to get `splitDetails` on `undefined`

```
let {splitDetails} = state
  .models
  .find(model => model.id === activeModelId)
|| {};
```
May be?